dashboard/app/models/concerns/user/level_progressable.rb

```diff
@@ -0,0 +1,156 @@
+module User::LevelProgressable
+  extend ActiveSupport::Concern
+
+  # Returns the next visible script_level for the next progression level in the # given script that hasn't yet been passed, starting at the last level the
```

Comment:
I see this line was already like this in `user.rb`, but if you're open to doing some additional cleanup on this code while you're in the area:

```suggestion
  # Returns the next visible script_level for the next progression level in the
  # given script that hasn't yet been passed, starting at the last level the
```